### PR TITLE
Access control

### PR DESCRIPTION
**Pull request by GuentherJulian**
Closes https://github.com/devonfw-sample/devon4quarkus-reference/issues/27
* Added Smallrye JWT extension
* Added role-based access control to the products service to secure the delete and post method
* Adapted tests
* Added documentation on how to deploy and configure keycloak

### CODE REVIEW

File: documentation/Keycloak/keycloak_authentication.asciidoc [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+* link:https://github.com/devonfw/devon4j/blob/master/documentation/guide-access-control.asciidoc[devon4j documentation]
+
+This guide shows how to enable role-based access control using JWTs (link:https://github.com/devonfw/devon4j/blob/master/documentation/guide-jwt.asciidoc[JSON Web Tokens]) and authentication using Keycloak as centralized IAM (Identity and Access Management) solution.
+
+== Setup and configure Keycloak
+
+The `docker-compose.yaml` file already includes a configuration for the setup of Keycloak. Use the `docker-compose up` command to deploy Keycloak in your Docker environment.
+You can also install Keycloak on your local machine without Docker. For this, follow the instructions on the official link:https://www.keycloak.org/docs/latest/server_installation/index.html[Keycloak website].
+
+You can access Keycloak by browsing to http://localhost:8180/auth/ in your browser. Open the Administration console and log in with the default username "admin" and password "admin".
```
Comment (rafaelvicariabarker) [inline documentation/Keycloak/keycloak_authentication.asciidoc:19]: by **navigating** to ... in your browser

File: documentation/Keycloak/keycloak_authentication.asciidoc [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+The first step is to create a new realm. A realm in Keycloak is used to manage a set of users with their credentials, groups and roles. Click on "Add realm" and then select the `keycloak-realm.json` file from the documentation/Keycloak folder of this repository.
+This will create the `devon4quarkus-product` realm with some preconfigured roles.
+
+In the next step, add a new user. Click on the "Users" section in the outline, then on "Add user" and fill in the form fields.
+
+image::keycloak_add_user.png[Add a new user]
+
+After this, open the "Credentials" tab and enter the credentials for the user. The newly created user is now able to authenticate himself in Keycloak. By default, new users must enter a new password themselves the first time they log in to fully set up the new account. So open  http://localhost:8180/auth/realms/devon4quarkus-product/account, sign in with the new user and enter new credentials for him.
```
Comment (rafaelvicariabarker) [inline documentation/Keycloak/keycloak_authentication.asciidoc:28]: ...enter the credentials **of** the user

File: documentation/Keycloak/keycloak_authentication.asciidoc [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  "client_id" = "devon4quarkus-product-cli"
+  "grant_type" = "password"
+  "username" = "john doe"
+  "password" = "demo"
+}
+
+Invoke-WebRequest -method POST -body $body -contenttype "application/x-www-form-urlencoded" http://localhost:8180/auth/realms/devon4quarkus-product/protocol/openid-connect/token | Select-Object -Expand content
+----
+
+Finally we can again test the delete method. Pass the "access_token" from the previous response as Bearer token in the Authorization header of the request.
```
Comment (rafaelvicariabarker) [inline documentation/Keycloak/keycloak_authentication.asciidoc:114]: Finally, we can test the delete method **again**.

**Review by rafaelvicariabarker — CHANGES_REQUESTED** [review]
Made some language suggestions

**Review by rafaelvicariabarker — APPROVED** [review]

**Comment by maybeec** [comment]
@GuentherJulian please check the credentials and please redo a clean PR. There are private keys  here as part of the PR.
Not sure where they come from. But we have to be careful to not have any passwords and private keys in Git if possible although if they are just for "testing" as it will ring alarms at different ends as well and we'll get contacted to clarify such things as there is always the risk that we will oversee some critical occurrences for sure.

**Review by rafaelvicariabarker — CHANGES_REQUESTED** [review]

**Pull request closed without merging**